long_tests/test_generic_cg_elastic.py

```diff
@@ -0,0 +1,195 @@
+                         options=dict(gtol=gtol, ftol=1e-20))
+
+    print(res.message, res.nfev)
+    assert res.success
+    CA_lbfgsb = res.x.reshape((nx, ny)) > 0  # Contact area
+    print("CA lbfgsb {}".format(CA_lbfgsb.sum() / (nx * ny)))
+    # print(CA_lbfgsb / (nx * ny))
+    _lbfgsb = res.x.reshape((nx, ny))
+    fun = system.dual_objective(offset, gradient=True)
+    gap_lbfgsb = fun(res.x)[1]
+    gap_lbfgsb = gap_lbfgsb.reshape((nx, ny))
+
+    # ###################BUGNICOURT########################################
+    bugnicourt_cg.constrained_conjugate_gradients(system.dual_objective
+                                                  (offset, gradient=True),
+                                                  system.
+                                                  dual_hessian_product,
+                                                  init_pressure,
+                                                  mean_val=None, gtol=gtol)
+    assert res.success
+
+    bugnicourt = res.x.reshape((nx, ny))
+    CA_bugnicourt = res.x.reshape((nx, ny)) > 0  # Contact area
+    gap_bugnicourt = fun(res.x)[1]
+    gap_bugnicourt = gap_bugnicourt.reshape((nx, ny))
+    #
+    # # ##################POLONSKY-KEER#####################################
+    res = polonsky_keer.min_cg(
+        system.dual_objective(offset, gradient=True),
+        system.dual_hessian_product,
+        init_pressure, gtol=gtol)
+    assert res.success
+
+    CA_polonsky = res.x.reshape((nx, ny)) > 0  # Contact area
+    gap_polonsky = fun(res.x)[1]
+    gap_polonsky = gap_polonsky.reshape((nx, ny))
+
+    np.testing.assert_allclose(gap_lbfgsb, gap_polonsky, atol=1e-3)
+    np.testing.assert_allclose(gap_lbfgsb, gap_bugnicourt, atol=1e-3)
+    np.testing.assert_allclose(gap_bugnicourt, gap_polonsky, atol=1e-3)
+
+    # ##########TEST MEAN VALUES#######################################
+    mean_val = np.mean(_lbfgsb)
+    print('mean {}'.format(mean_val))
+    # ####################POLONSKY-KEER##############################
+    res = polonsky_keer.min_cg(
+        system.dual_objective(offset, gradient=True),
+        system.dual_hessian_product,
+        init_pressure, mean_value=mean_val, gtol=gtol)
+
+    assert res.success
+    polonsky_mean = res.x.reshape((nx, ny))
+
+    # # ####################BUGNICOURT###################################
+    bugnicourt_cg.constrained_conjugate_gradients(system.dual_objective
+                                                  (offset, gradient=True),
+                                                  system.
+                                                  dual_hessian_product,
+                                                  init_pressure,
+                                                  mean_val=mean_val,
+                                                  gtol=gtol)
+    assert res.success
+
+    bugnicourt_mean = res.x.reshape((nx, ny))
+
+    np.testing.assert_allclose(polonsky_mean, _lbfgsb, atol=1e-3)
```

Comment:
please consider an other name. Like above, gap_polonsky.  The variable name suggests that it the mean value of something but it is actually the array of the gaps.